rtnl/route.go

```diff
@@ -0,0 +1,30 @@
+// github.com/jsimonetti/rtnetlink/rtnl
+package rtnl
+
+import (
+	"github.com/jsimonetti/rtnetlink"
+	"golang.org/x/sys/unix"
+	"net"
+)
+
+func RouteAdd(conn *rtnetlink.Conn, ifc *net.Interface, dst net.IPNet, gw net.IP) (err error) {
+	prefixlen, _ := dst.Mask.Size()
+	attr := rtnetlink.RouteAttributes{
+		Dst:      dst.IP,
+		OutIface: uint32(ifc.Index),
+	}
+	if gw != nil {
+		attr.Gateway = gw
+	}
+	tx := &rtnetlink.RouteMessage{
+		Family:     unix.AF_INET,
+		Table:      unix.RT_TABLE_MAIN,
+		Protocol:   unix.RTPROT_BOOT,
```

Comment:
to be checked: Table, Protocol, Scope and Type should be probably inferred

Reply:
Checked against [iproute2](https://github.com/shemminger/iproute2/blob/master/ip/iproute.c#L1085). Scope/table can be overridden on command line but the defaults look same as we do here. I think we're good here for most uses.